src/main/java/sirius/db/mixing/EntityDescriptor.java

```diff
@@ -405,16 +407,20 @@ public boolean isFetched(BaseEntity<?> entity, Property property) {
     /**
      * Determines if the value for the property was changed since it was last fetched from the database.
      * <p>
-     * For {@link LocalDateTimeProperty}, we consider seconds as the smallest unit for comparison,
-     * fully ignoring milliseconds.
+     * For {@link LocalDateTimeProperty}, we consider milliseconds as the smallest unit for comparison,
+     * ignoring micro- and nanoseconds.
      *
      * @param entity   the entity to check
      * @param property the property to check for
      * @return <tt>true</tt> if the value was changed, <tt>false</tt> otherwise
      */
     public boolean isChanged(BaseEntity<?> entity, Property property) {
         if (property instanceof LocalDateTimeProperty) {
-            return isChanged(entity, property, this::isLocalDateTimeEqual);
+            if (entity instanceof SQLEntity) {
+                return isChanged(entity, property, (a, b) -> isLocalDateTimeEqual(a, b, ChronoUnit.SECONDS));
+            } else {
+                return isChanged(entity, property, (a, b) -> isLocalDateTimeEqual(a, b, ChronoUnit.MILLIS));
+            }
```

Comment:
Would be more clean if we could put this into the DB specific stuff, but I don't know if there is a place where this fits nicely.